runtests.py

```diff
@@ -183,7 +183,7 @@ def main(argv):
     if args.shell:
         shell = os.environ.get('SHELL', 'sh')
         print("Spawning a Unix shell...")
-        os.execv(shell, [shell] + extra_argv)
+        subprocess.check_call([shell] + extra_argv)
         sys.exit(1)
```

Comment:
What I'm concerned about is that `sys.exit(1)` is an error return, whereas the `check_call` function raises and exception on error and returns normally otherwise. So perhaps `sys.exit()` instead?